### PR TITLE
feat(column-group): add support of column group into ob-sql-parser 

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:

add support of column group into ob-sql-parser .
There are three grammars related to column group, which are:
1. create table
2. create index
3. alter table

### CODE REVIEW

File: libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/oracle/OracleExpressionFactory.java [hunk 1]
```diff
@@ -916,6 +917,25 @@ public Expression visitJson_table_expr(Json_table_exprContext ctx) {
...
     }
 
+    public Expression visitJson_equal_expr(Json_equal_exprContext ctx) {
+        List<FunctionParam> params = new ArrayList<>();
+        if (ctx.func_param_list() != null) {
+            params.addAll(ctx.func_param_list().func_param().stream()
+                    .map(this::visitFunctionParam).collect(Collectors.toList()));
+        }
+        FunctionCall fCall = new FunctionCall(ctx, ctx.getChild(0).getText(), params);
+        if (ctx.json_equal_option() != null) {
```
Comment (yhilmare) [inline libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/oracle/OracleExpressionFactory.java:927]: we may extract a method to construct JsonOnOption

File: libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/oracle/OracleExpressionFactory.java [hunk 2]
```diff
@@ -1218,6 +1240,7 @@ public FunctionCall getFunctionCall(Access_func_exprContext ctx) {
             } else {
                 jsonOnOption.setOnError(new ConstExpression(ctx.json_equal_option().ERROR_P(0)));
             }
+            fCall.addOption(jsonOnOption);
```
Comment (yhilmare) [inline libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/oracle/OracleExpressionFactory.java:1243]: great, 👍

**Comment by yhilmare** [comment]
with column group lacks implementation on out_of_line_index and out_of_line_constraint

File: libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/statement/common/ColumnGroupElement.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    public ColumnGroupElement(ParserRuleContext ruleNode, String groupName, List<String> columnNames) {
+        super(ruleNode);
+        this.groupName = groupName;
+        this.columnNames = columnNames;
+    }
+
+    @Override
+    public String toString() {
+        if (isAllColumns()) {
+            return "all columns";
```
[the remark this answers is not in the document]
Comment (LuckyPickleZZ) [reply libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/statement/common/ColumnGroupElement.java]: done

File: libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/statement/common/ColumnGroup.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        this.columnGroupElements = columnGroupElements;
+    }
+
+    public ColumnGroup(List<ColumnGroupElement> columnGroupElements) {
+        this.columnGroupElements = columnGroupElements;
+    }
+
+    @Override
+    public String toString() {
+        return " COLUMN GROUP(" +
```
[the remark this answers is not in the document]
Comment (LuckyPickleZZ) [reply libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/statement/common/ColumnGroup.java]: `ColumnGroup` class was removed.

File: libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/statement/createindex/CreateIndex.java [hunk 5]
```diff
@@ -52,6 +53,7 @@ public class CreateIndex extends BaseStatement {
     private IndexOptions indexOptions;
     private Partition partition;
     private final List<SortColumn> columns;
+    private List<ColumnGroupElement> columnGroupElements;
```
[the remark this answers is not in the document]
Comment (LuckyPickleZZ) [reply libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/statement/createindex/CreateIndex.java:56]: `ColumnGroup` class was removed.

File: libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/mysql/MySQLAlterTableActionFactory.java [hunk 6]
```diff
@@ -234,6 +241,20 @@ public AlterTableAction visitAlter_constraint_option(Alter_constraint_optionCont
         return action;
     }
 
+    @Override
+    public AlterTableAction visitAlter_column_group_option(Alter_column_group_optionContext ctx) {
+        AlterTableAction action = new AlterTableAction(ctx);
+        List<ColumnGroupElement> columnGroupElements = ctx.column_group_list().column_group_element()
+                .stream().map(c -> new MySQLColumnGroupElementFactory(c).generate()).collect(Collectors.toList());
+        ColumnGroup columnGroup = new ColumnGroup(ctx, columnGroupElements);
```
[the remark this answers is not in the document]
Comment (LuckyPickleZZ) [reply libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/mysql/MySQLAlterTableActionFactory.java]: `ColumnGroup` class was removed.

File: libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/mysql/MySQLColumnGroupElementFactory.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        if (ctx.ALL() != null) {
+            return new ColumnGroupElement(ctx, true, false);
+        } else if (ctx.EACH() != null) {
+            return new ColumnGroupElement(ctx, false, true);
+        }
+        if (ctx.relation_name() == null || ctx.column_name_list() == null) {
+            throw new IllegalStateException("Missing column group");
+        }
+        Relation_nameContext relationNameContext = ctx.relation_name();
+        String relationName = relationNameContext.getText();
```
[the remark this answers is not in the document]
Comment (LuckyPickleZZ) [reply libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/mysql/MySQLColumnGroupElementFactory.java]: done~

File: libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/oracle/OracleColumnGroupElementFactory.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    public ColumnGroupElement visitColumn_group_element(Column_group_elementContext ctx) {
+        if (ctx.ALL() != null) {
+            return new ColumnGroupElement(ctx, true, false);
+        } else if (ctx.EACH() != null) {
+            return new ColumnGroupElement(ctx, false, true);
+        }
+        if (ctx.relation_name() == null || ctx.column_name_list() == null) {
+            throw new IllegalStateException("Missing column group");
+        }
+        Relation_nameContext relationNameContext = ctx.relation_name();
```
[the remark this answers is not in the document]
Comment (LuckyPickleZZ) [reply libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/oracle/OracleColumnGroupElementFactory.java]: `ColumnGroup` class was removed.

**Comment by LuckyPickleZZ** [comment]
> with column group lacks implementation on out_of_line_index and out_of_line_unique_index

done

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**